src/widgets/menu.rs

```diff
@@ -0,0 +1,192 @@
+use gdk;
+use glib::translate::*;
+use traits::FFIWidget;
+use ffi;
+
+struct_Widget!(Menu);
+impl_TraitWidget!(Menu);
+
+impl Menu {
+    pub fn new() -> Option<Menu> {
+        let tmp_pointer = unsafe { ffi::gtk_menu_new() };
+        check_pointer!(tmp_pointer, Menu)
+    }
+
+    //#[cfg(gtk_3_4)]
+    //pub fn new_from_model<T: Upcast<gio::MenuModel>>(model: &T) -> Menu {
+    //    unsafe { TODO: call ffi:gtk_menu_new_from_model() }
+    //}
+
+    pub fn attach<T: ::WidgetTrait>(&self, child: &T, left_attach: u32, right_attach: u32, top_attach: u32, bottom_attach: u32) {
+        unsafe {
+            ffi::gtk_menu_attach(GTK_MENU(self.unwrap_widget()), child.unwrap_widget(), left_attach, right_attach, top_attach, bottom_attach);
+        }
+    }
+
+    //pub fn attach_to_widget<T: ::WidgetTrait>(&self, attach_widget: &T, detacher: /*Unknown kind*/Unknown rust type: "MenuDetachFunc") {
+    //    unsafe { TODO: call ffi:gtk_menu_attach_to_widget() }
+    //}
+
+    pub fn detach(&self) {
+        unsafe {
+            ffi::gtk_menu_detach(GTK_MENU(self.unwrap_widget()));
+        }
+    }
+
+    // pub fn get_accel_group(&self) -> Option<AccelGroup> {
+    //     unsafe {
+    //         from_glib_none(ffi::gtk_menu_get_accel_group(GTK_MENU(self.unwrap_widget())))
+    //     }
+    // }
+
+    pub fn get_accel_path(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gtk_menu_get_accel_path(GTK_MENU(self.unwrap_widget())))
+        }
+    }
+
+    // pub fn get_active<T: ::WidgetTrait>(&self) -> Option<T> {
+    //     unsafe {
+    //         from_glib_none(ffi::gtk_menu_get_active(GTK_MENU(self.unwrap_widget())))
+    //     }
+    // }
+
+    // pub fn get_attach_widget<T: ::WidgetTrait>(&self) -> Option<T> {
+    //     unsafe {
+    //         from_glib_none(ffi::gtk_menu_get_attach_widget(GTK_MENU(self.unwrap_widget())))
+    //     }
+    // }
+
+    pub fn get_monitor(&self) -> i32 {
+        unsafe {
+            ffi::gtk_menu_get_monitor(GTK_MENU(self.unwrap_widget()))
+        }
+    }
+
+    pub fn get_reserve_toggle_size(&self) -> bool {
+        unsafe {
+            from_glib(ffi::gtk_menu_get_reserve_toggle_size(GTK_MENU(self.unwrap_widget())))
+        }
+    }
+
+    pub fn get_tearoff_state(&self) -> bool {
+        unsafe {
+            from_glib(ffi::gtk_menu_get_tearoff_state(GTK_MENU(self.unwrap_widget())))
+        }
+    }
+
+    pub fn get_title(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gtk_menu_get_title(GTK_MENU(self.unwrap_widget())))
+        }
+    }
+
+    pub fn popdown(&self) {
+        unsafe {
+            ffi::gtk_menu_popdown(GTK_MENU(self.unwrap_widget()));
+        }
+    }
+
+    // pub fn popup<T: ::WidgetTrait, U: ::WidgetTrait, V>(&self, parent_menu_shell: Option<&T>, parent_menu_item: Option<&U>, func: fn(menu: &Menu, x: &mut i32, y: &mut i32, push_in: &mut bool, data: &mut V), data: V, button: u32, activate_time: u32) {
```

Comment:
We could go for the easier option here first: pass `None` as `func` and `ptr::null_mut()` as `data`. The positioning function has a reasonable default doesn't it?